repos/spack_repo/builtin/packages/texlive/package.py

```diff
@@ -155,7 +158,18 @@ def setup_texlive(self):
         with working_dir("spack-build"):
             make("texlinks")
 
-        copy_tree("texlive-{0}-texmf".format(self.version.string), self.prefix)
+        ignore_doc = "~doc" in self.spec
+        ignore_src = "~src" in self.spec
+
+        ignore = lambda f: (
+            len(f.split(os.sep)) > 1
+            and (
+                (ignore_doc and f.split(os.sep)[1] == "doc")
+                or (ignore_src and f.split(os.sep)[1] == "source")
+            )
```

Comment:
Seems wasteful to do `f.split(os.sep)` up to three times per inode.

Reply:
Nevermind, the rest of the tree is not even visited once the parents get ignored.